ncoreparser/parser.py

```diff
@@ -25,24 +30,43 @@ def get_key(data):
         raise NcoreParserError(f"Error while read user "
                                f"key with pattern: {key_pattern}")
 
+
+    def id_exists(self, data, search_id):
+        for i, id in enumerate(data):
+            if search_id in data[i]:
+                return True
+
+        return False
+
+
     def get_items(self, data):
         types = self.type_pattern.findall(data)
-        ids_names = self.id_name_pattern.findall(data)
+        ids_names_posters = self.id_name_poster_pattern.findall(data)
+        ids_names = self.id_name_patter.findall(data)
         dates_times = self.date_pattern.findall(data)
         sizes = self.size_pattern.findall(data)
         seed = self.seeders_pattern.findall(data)
         leech = self.leechers_pattern.findall(data)
-        if len(types) != 0 and len(types) == len(ids_names) == \
```

Comment:
You can create a poster dict here, where the key is the torrent id and the value is the poster url or "no cover".
You can iterate on IDs, and add the id to this dict key, then find the poster image and add as value.

For this I recommend to convert the re.findall response to a dictionary:
https://www.geeksforgeeks.org/python-convert-a-list-of-tuples-into-dictionary/